### ISSUE TITLE
Add Random Events

### ISSUE THREAD

**Issue by lmachens** [lmachens/soc.gg#89]
Example:

Faction: **Arleon**
Name: **Prize Hen**
Description: **A patrol stumble upon a village in celebration. They quickly find out what the ruckus is all about. A hen has layed a solid golden egg. The egg and the hen were swiftly claimed for the realm's profit.**

In addition, details about triggers, recipient value, artefact types or penalities would be nice.

Full data extracted from game:

```json
{
    "id": 2,
    "randomEventData": {
      "type": 1,
      "uniqueName": "Arleon/GoldenEgg/PrizeHen",
      "nameKey": "RandomEvents/Arleon/GoldenEgg/PrizeHen",
      "descriptionKey": "RandomEvents/Arleon/GoldenEgg/PrizeHen/Description",
      "eventChainNameKey": "RandomEvents/Arleon/GoldenEgg",
      "isReoccuring": 0,
      "chanceOfHappening": 0.08,
      "eventEvaluationTrigger": 0,
      "requirementEvaluationType": 0,
      "requirements": [
        {
          "requirementType": 7,
          "resourceType": 0,
          "entityType": 6,
          "artifactType": 7,
          "factionIndex": 0,
          "unitIndex": 0,
          "eventReference": {
            "uniqueName": null
          },
          "value": 5
        }
      ],
      "eventRecipient": 0,
      "recipientArtifactType": 7,
      "recipientFactionIndex": 1,
      "recipientUnitIndex": 1,
      "recipientTroopUpgradeType": 0,
      "recipientValue": 500,
      "reward": {
        "rewardDataList": []
      },
      "penalty": {
        "penaltyDataList": [
          {
            "penaltyType": 0,
            "resourcePenalty": {
              "type": 0,
              "amountMinMax": {
                "min": 500,
                "max": 800
              }
            },
            "destroyOwnedBuilding": {
              "buildingToDestroy": 0,
              "destroyAll": 0,
              "amount": 1
            },
            "createHostile": {
              "troop1": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop2": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop3": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop4": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop5": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop6": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop7": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop8": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "troop9": {
                "factionIndex": 0,
                "unitIndex": 0,
                "size": 0,
                "upgradeType": 0
              },
              "amountOfHostiles": 1,
              "spawnLocation": 0,
              "battleMapName": null
            },
            "createRandomHostile": {
              "faction": {
                "factionIndex": 1
              },
              "tier": 1,
              "maxUpgradeType": 0,
              "maxCost": 5000,
              "troopCount": 5,
              "amountOfHostiles": 1,
              "spawnLocation": 0,
              "battleMapName": null
            },
            "reduceRecruitmentPool": {
              "troopsToReduce": [],
              "percentage": 0
            }
          }
        ]
      }
    },
    "type": "GoldenEggPrizeHen"
  }
  ```

**Comment by rizfn** [comment]
Hi, I'm willing to work on this

**Comment by lmachens** [comment]
Solved in #92
